packages/eslint-plugin-query/src/rules/exhaustive-deps/exhaustive-deps.test.ts

```diff
@@ -236,6 +236,17 @@ ruleTester.run('exhaustive-deps', rule, {
         }
       `,
     },
+    {
+      name: 'should not fail if queryKey is having the whole object while queryFn uses some props of it',
+      code: normalizeIndent`
+        const state = { foo: 'foo', bar: 'bar' }
+    
+        useQuery({
+            queryKey: ['state', state],
+            queryFn: () => Promise.resolve({ foo: state.foo, bar: state.bar })
+        })
```

Comment:
will this fail if state has more options than what is used in the `queryFn`, like:

```
const state = { foo: 'foo', bar: 'bar', baz: 'baz' }
```

as stated in one of my comments, this would mean duplicate cache entries if we only use `foo` and `bar` in the `queryFn`, but `baz` changes values. It would be cool if those cases error as well, like:

```
const useFoo = (id) => useQuery({ queryKey: ['foo', id], queryFn: () => fetchFoo() })
```

also here, `id` is unused and probably _shouldn't_ be in the key. It's a different problem that "exhaustiveness", because it goes in the other direction. Probably nothing that blocks this PR, but I'd be curious if this is something that we already cover, or maybe should cover ...

Reply:
Currently, the plugin doesn't error on "unusedDeps". I wrote a small sketch that should mitigate most of the cases, but it currently doesn't support your example (but it could be).

Although, what if `state` can be one nullable/optional? or what if it's a union of multiple types?

Reply:
yeah it gets difficult for sure. gonna merge this one 👍